src/Nethermind/Nethermind.Evm/Precompiles/AddressExtensions.cs

```diff
@@ -2,47 +2,40 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Runtime.InteropServices;
 using Nethermind.Core;
-using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
 
-namespace Nethermind.Evm.Precompiles
+namespace Nethermind.Evm.Precompiles;
+
+public static class AddressExtensions
 {
-    public static class AddressExtensions
+    public static bool IsPrecompile(this Address address, IReleaseSpec releaseSpec)
     {
-        private static byte[] _nineteenZeros = new byte[19];
-
-        public static bool IsPrecompile(this Address address, IReleaseSpec releaseSpec)
+        Span<int> data = MemoryMarshal.Cast<byte, int>(address.Bytes.AsSpan());
+        return data[0] == 0 && data[1] == 0 && data[2] == 0 && data[3] == 0
```

Comment:
Reverse order of 0 - 4; the the Jit will only bounds check `data[4]` access and know all the others are fine and skip